instrumentation/aws-sdk/aws-sdk-1.11/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v1_11/AwsExperimentalAttributes.java

```diff
@@ -17,7 +17,7 @@ final class AwsExperimentalAttributes {
   static final AttributeKey<String> AWS_QUEUE_NAME = stringKey("aws.queue.name");
   static final AttributeKey<String> AWS_STREAM_NAME = stringKey("aws.stream.name");
   static final AttributeKey<String> AWS_TABLE_NAME = stringKey("aws.table.name");
-  static final AttributeKey<String> AWS_REQUEST_ID = stringKey("aws.requestId");
+  static final AttributeKey<String> AWS_REQUEST_ID = stringKey("aws.request_id");
```

Comment:
just a small note, i tried to replicate this attribute in the 1.11 tests so I could update the assertions to account for it, but as far as I can tell the [code path to add this attribute](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/aws-sdk/aws-sdk-1.11/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v1_11/AwsSdkExperimentalAttributesExtractor.java#L63-L67) is never executed. 

It seems as if in the tests the `response.getAwsResouce()` is never an instance of `AmazonWebServiceResponse`, which makes sense for `com.amazonaws:aws-java-sdk-sqs:1.11.0` since the request's [don't extend it](https://github.com/aws/aws-sdk-java/blob/1.11.0/aws-java-sdk-sqs/src/main/java/com/amazonaws/services/sqs/model/CreateQueueResult.java#L26), but even testing with `1.11.106` [which does extend it](https://github.com/aws/aws-sdk-java/blob/1.11.106/aws-java-sdk-sqs/src/main/java/com/amazonaws/services/sqs/model/CreateQueueResult.java#L27), and I still wasn't able to observe this attribute being added.

Reply:
looks like the result of `response.getAwsResponse()` is instance of `AmazonWebServiceResult` not `AmazonWebServiceResponse`. Perhaps it should use
```java
    if (response != null && response.getAwsResponse() instanceof AmazonWebServiceResult) {
      AmazonWebServiceResult<?> awsResp = (AmazonWebServiceResult<?>) response.getAwsResponse();
      String requestId = awsResp.getSdkResponseMetadata().getRequestId();
      if (requestId != null) {
        attributes.put(AWS_REQUEST_ID, requestId);
      }
    }
```

Reply:
oh wow yea you are right, thank you for pointing that out

Reply:
it looks like the `AmazonWebServiceResult` class was first introduced in [1.11.33](https://github.com/aws/aws-sdk-java/commits/master/aws-java-sdk-core/src/main/java/com/amazonaws/AmazonWebServiceResult.java), so to fix the muzzle failures we  would need to bump the [versions in the configs](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/aws-sdk/aws-sdk-1.11/javaagent/build.gradle.kts#L16) from `1.10.33` to `1.11.33`, is that an acceptable thing to do in this case?

Reply:
Have a look at https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/VERSIONING.md#dropping-support-for-older-library-versions
When it is easy to avoid bumping the minimum supported version, like in this case, we usually just use a bit or reflection or `@NoMuzzle` to get around this. Something like
```java
  private static final boolean CAN_GET_RESPONSE_METADATA = canGetResponseMetadata();

  private static boolean canGetResponseMetadata() {
    try {
      Class<?> clazz = Class.forName("com.amazonaws.AmazonWebServiceResult");
      clazz.getMethod("getSdkResponseMetadata");
      return true;
    } catch (ClassNotFoundException | NoSuchMethodException exception) {
      return false;
    }
  }

  @Override
  public void onEnd(
      AttributesBuilder attributes,
      Context context,
      Request<?> request,
      @Nullable Response<?> response,
      @Nullable Throwable error) {
    ResponseMetadata responseMetadata = getResponseMetadata(response);
    if (responseMetadata != null) {
      String requestId = responseMetadata.getRequestId();
      if (requestId != null) {
        attributes.put(AWS_REQUEST_ID, requestId);
      }
    }
  }

  @NoMuzzle
  private static ResponseMetadata getResponseMetadata(Response<?> response) {
    if (CAN_GET_RESPONSE_METADATA
        && response != null
        && response.getAwsResponse() instanceof AmazonWebServiceResult) {
      AmazonWebServiceResult<?> awsResp = (AmazonWebServiceResult<?>) response.getAwsResponse();
      return awsResp.getSdkResponseMetadata();
    }
    return null;
  }
```
Instead of `@NoMuzzle` could use reflection to make the one method call that is not present in older versions.
Bumping the minimum supported version of aws-sdk1.1 has come up before in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8889